### DISCUSSION TITLE
Reducing CI load with automerge and rebaseWhen

### DISCUSSION THREAD

**Discussion by adam-moss** [renovatebot/renovate#29761] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

latest official via docker

### Please tell us more about your question or problem

When a repository is configured for fast-forward merges and thus renovate with `rebaseWhen: "behind-base-branch"` and `automergeType: branch` what would be the preferred configuration to reduce CI load without unduly impacting flow?

For example say a repository has the default max of 10 concurrent and they're all eligible for automerge.

At the moment what happens is renovate merges 1 and rebases 9, so 10 pipelines.  Of these 8 are effectively redundant.

On next run (assuming no other changes) it'll merge 1, rebase 8, so 9 pipelines of which 7 are redundant.

From an operational point of view only 2 pipelines are actually required in this scenario, the one being merged and the one being rebased to merge next, the others could conceivably be pushed with `--push-option=ci.skip` so they're updated but not "passing".

I'd be interested in ideas for minimising this overhead whilst not impacting non-automerge flows?

Thanks.

### Logs (if relevant)

n/a

**Comment by HonkingGoose** [comment]
If you're on GitHub: try GitHub's Merge Queue feature. Read the GitHub docs for that first, and then search the Renovate docs with the keyword Merge Queue.

I think GitLab has Merge Trains, but I think Renovate does not support that (yet). Search the Renovate issues for Merge Trains.

The big idea behind Merge Queue and Merge Trains is that you can put PRs in a queue, and then let the Git-hosting platform worry about merging it. The platform will only merge PRs that pass the tests, and that work well with the other changes.

The [Renovate docs, Noise Reduction](https://docs.renovatebot.com/noise-reduction/) also has strategies to lower the total number of PRs that you get.

**Comment by rarkins** [comment, marked as answer]
Related: https://github.com/renovatebot/renovate/issues/12448
